app/src/main/java/com/babylon/wallet/android/domain/usecases/AuthorizeSpecifiedPersonaUseCase.kt

```diff
@@ -195,21 +194,17 @@ class AuthorizeSpecifiedPersonaUseCase @Inject constructor(
             ongoingAccounts = selectedAccounts.map { it.data },
             ongoingSharedPersonaData = selectedPersonaData
         ).mapCatching { response ->
-            return when (
-                dAppMessenger.sendWalletInteractionSuccessResponse(
-                    remoteConnectorId = request.remoteConnectorId,
-                    response = response
+            return dAppMessenger.sendWalletInteractionSuccessResponse(
+                remoteConnectorId = request.remoteConnectorId,
+                response = response
+            ).getOrNull()?.let {
+                val updatedDapp = updateDappPersonaWithLastUsedTimestamp(authorizedDapp, persona.address)
+                dAppConnectionRepository.updateOrCreateAuthorizedDApp(updatedDapp)
+                Result.success(
+                    authorizedDapp.displayName
                 )
-            ) {
-                is ResultInternal.Success -> {
-                    val updatedDapp = updateDappPersonaWithLastUsedTimestamp(authorizedDapp, persona.address)
-                    dAppConnectionRepository.updateOrCreateAuthorizedDApp(updatedDapp)
-                    Result.success(
-                        authorizedDapp.displayName
-                    )
-                }
-
-                else -> Result.failure(DappRequestFailure.InvalidRequest)
+            } ?: run {
```

Comment:
Same here no need for run but just `?: Result.failure(DappRequestFailure.InvalidRequest)`